File: Subprogramas/Polimorfismo/Polimorfismo.cpp
//C++

#include <iostream>

using namespace std;

class Polimorfismo {
public:
    void imprimir(const string& s) {
        cout << s << std::endl;
    }

    void imprimir(int i) {
        cout << i << endl;
    }
};

int main() {
    Polimorfismo p;
    p.imprimir("Hello");  // chama o método que aceita string
    p.imprimir(10);       // chama o método que aceita int
    return 0;
}

File: Subprogramas/Polimorfismo/Polimorfismo.cs
//C Sharp

using System;

class Polimorfismo {
    public void Imprimir(string s) {
        Console.WriteLine(s);
    }

    public void Imprimir(int i) {
        Console.WriteLine(i);
    }
}

class Program {
    static void Main() {
        Polimorfismo p = new Polimorfismo();
        p.Imprimir("Hello");  // chama o método que aceita string
        p.Imprimir(10);       // chama o método que aceita int
    }
}
